src/tablestore.rs

```diff
@@ -361,19 +390,22 @@ impl TableStore {
             object_store: self.object_store.clone(),
             path,
         };
-
         // Initialize the result vector and a vector to track uncached ranges
         let mut blocks_read = VecDeque::with_capacity(blocks.end - blocks.start);
         let mut uncached_ranges = Vec::new();
+        // currently blocks are written from index 0.
+        let block_offset = 0;
 
         // If block cache is available, try to retrieve cached blocks
         if let Some(cache) = &self.block_cache {
             // Attempt to get all requested blocks from cache concurrently
-            let cached_blocks =
-                join_all(blocks.clone().map(|block_num| async move {
-                    cache.get((handle.id, block_num)).await.block()
-                }))
-                .await;
+            let cached_blocks = join_all(blocks.clone().map(|block_num| async move {
+                cache
+                    .get((handle.id, block_num + block_offset))
```

Comment:
I'm not 100% clear this works as we expect it to. Does `block_num` represent the block's index or its offset? I think it might be the index ("block 0", "block 1", and so on). I think we want to use the block's offset here.

Reply:
Yea, it looks like `SsTableFormat.read_blocks()` converts the `blocks` range from an index to an offset:

```rust
        for block in blocks {
            let block_meta = index.block_meta().get(block);
            let block_bytes_start = block_meta.offset() as usize - start_offset;
            let block_bytes = if block == index.block_meta().len() - 1 {
                bytes.slice(block_bytes_start..)
            } else {
                let next_block_meta = index.block_meta().get(block + 1);
                let block_bytes_end = next_block_meta.offset() as usize - start_offset;
                bytes.slice(block_bytes_start..block_bytes_end)
            };
            decoded_blocks.push_back(self.decode_block(block_bytes, compression_codec.into())?);
        }
```

So I think we can't use `block_num` here. Instead, we should get the block's actual offset and use that (and no need for the `block_offset` variable you have now).

Reply:
The arg is a block num range right, so block_num + block_offset would be the actual block offset?